designs/osc_containerization/OSC-containerization.md

```diff
@@ -0,0 +1,90 @@
+# OSC Containerization
+
+We want to deploy OSC as a container i.e as a docker image. This allows us to deploy OSC service along with other similar services on a single host. This also allows easier management of OSC lifecycle(Upgrades, HA etc)
+
+## Assignees
+Arvind Nadendla https://github.com/arvindn05
+
+## Background
+
+OSC containerization for deployment allows greater flexibility on mixing services on same host.
+
+For example, in case of openstack we can co-locate osc on the controller node with the other controller services like keystone, nova etc. We can follow the same lifecycle as those services and utilize similar mechanisms for upgrades, data persistence, HA etc
+
+It also simplifies packaging and distribution.
+
+## Constraints and Assumptions
+
+##### Constraint
+- Support migrating current OSC installation to container based installation.
+
+##### Assumption
+- OSC will be deployed as a container only. The Virtual Machine appliance model will not be supported going forward.
+- For migration of VM-based OSC to container based OSC, some manual steps maybe required. Scripts should be provided in this case.
+
+## Design Changes
+
+##### Persistent data
+Persistent data stored by OSC will need to be decoupled and managed outside the lifecycle of the container.
+
+The following data will need to be decoupled from the lifecycle of the OSC container:
+- H2 database
+- Keystore
+- Truststore
+- Plugins
+- Service Function catalog image files
+- Log file
+- vmidcServer.conf
+
+This data would need to be exposed in docker volumes to the OSC container.
+
+##### Life cycle management
+Remove restart and shutdown operations possbile via OSC UI
+Remove inplace upgrade of OSC via server upgrade bundles
+
+##### CLI/Restricted Shell
+Remove restricted shell. Restricted shell is no longer applicable as we will be using standard base containers.
+Note: It is technically feasible to have a restricted shell if we build up the OSC container image from scratch. For now, this is out of scope.
+
+##### Network/Ip address management
```

Comment:
I presume we can still run all the scenarios of OSC (like creation VC,MC,DA,DS, SG binding etc) with changes proposed in the networking/ip addressing management.

Reply:
yes.